sr_description/hand/xacro/thumb/thmiddle.gazebo.xacro

```diff
@@ -1,25 +1,25 @@
 <?xml version="1.0"?>
 <robot xmlns:xacro="http://www.ros.org/wiki/xacro">
-  <xacro:macro name="thmiddle_gazebo" params="prefix">
-
-  <!-- prefix can be anything but usually is "r_" or  "l_" for right and left hands distinction -->
-  <gazebo reference="${prefix}thmiddle">
-      <sensor type="contact" name="${prefix}thmiddle_contact">
-        <always_on>1</always_on>
-        <update_rate>1000.0</update_rate>
-        <contact>
-          <collision>${prefix}thmiddle_collision</collision>
-          <topic>${prefix}thmiddle_collision</topic>
-        </contact>
-        <plugin name="${prefix}thmiddle_gazebo_ros_bumper_controller" filename="libgazebo_ros_bumper.so">
-          <alwaysOn>true</alwaysOn>
-          <frameName>${prefix}thmiddle</frameName>
-          <updateRate>100.0</updateRate>
-          <bumperTopicName>contacts/${prefix}th/middle</bumperTopicName>
-        </plugin>
-      </sensor>
-      <selfCollide>true</selfCollide>
-       
+  <xacro:macro name="thmiddle_gazebo" params="prefix:=^ mid_sensor:=^|std">
+    <!-- prefix can be anything but usually is "r_" or  "l_" for right and left hands distinction -->
+    <gazebo reference="${prefix}thmiddle">
+      <xacro:unless value="${'none' in mid_sensor}">
+        <sensor type="contact" name="${prefix}thmiddle_contact">
+          <always_on>1</always_on>
+          <update_rate>1000.0</update_rate>
+          <contact>
+            <collision>${prefix}thmiddle_collision</collision>
+            <topic>${prefix}thmiddle_collision</topic>
+          </contact>
+          <plugin name="${prefix}thmiddle_gazebo_ros_bumper_controller" filename="libgazebo_ros_bumper.so">
+            <alwaysOn>true</alwaysOn>
+            <frameName>${prefix}thmiddle</frameName>
+            <updateRate>100.0</updateRate>
+            <bumperTopicName>contacts/${prefix}th/middle</bumperTopicName>
+          </plugin>
+        </sensor>
+        <selfCollide>true</selfCollide>
```

Comment:
@guihomework: This `selfCollide` tag is part of the `unless` clause here, while in all other places it's outside.
Is this one wrong here?

Reply:
yes it makes no sense to remove the selfcollide here and not at other places, yet another mistake of mine. 

side note : I am not convinced anymore by the current set of true/false on the self collide in the fingers, but I think it is a tricky matter (at least I remember it was not obvious) and if the current set works one should not try change it.